DynamicTablesPkg/Include/ArchCommonNameSpaceObjects.h

```diff
@@ -652,6 +656,41 @@ typedef struct CmArchCommonPccSubspaceType5Info {
 */
 typedef AML_PSD_INFO CM_ARCH_COMMON_PSD_INFO;
 
+/** A structure that describes TPM interface and access method.
+
+  TCG ACPI Specification 2.0
+
+  ID: EArchCommonObjTpm2InterfaceInfo
+*/
+typedef struct CmArchCommonTpm2InterfaceInfo {
+  /** Platform Class
+        0: Client platform
+        1: Server platform
+  */
+  UINT16    PlatformClass;
+
+  /** Physical address of the Control Area */
+  UINT64    AddressOfControlArea;
+
+  /** The Start Method selector determines which mechanism the
+      device driver uses to notify the TPM 2.0 device that a
+      command is available for processing.
+  */
+  UINT32    StartMethod;
+
+  /** Start method specific parameters */
+  UINT8     StartMethodParameters[EFI_TPM2_ACPI_TABLE_START_METHOD_SPECIFIC_PARAMETERS_MAX_SIZE];

```

Comment:
Minor: Is it possible to place the StartMethodParametersSize field before the StartMethodParameters field, please?
The reason I say this is in the future we may extend the ConfigurationManager Object parser to extract the StartMethodParametersSize value and then use it to print the StartMethodParameters. Something similar to what we do today in Acpiview.

Other than that this patch series looks good to me.

Reply:
I have swapped the Size field. Thanks.